### PR TITLE
Add support for ALLOW_UNTRUSTED env var

### PR DESCRIPTION
**Pull request by marcphilipp**

### CODE REVIEW

File: components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java [hunk 1]
```diff
@@ -48,22 +55,49 @@ public GradleEnterpriseApiClient(URL baseUrl, CustomValueNames customValueNames,
...
+
+        if ("1".equals(System.getenv(ALLOW_UNTRUSTED))) {
+            try {
+                SSLContext sslContext = SSLContext.getInstance("SSL");
+                sslContext.init(null, new TrustManager[]{TRUST_ALL_CERTS}, new java.security.SecureRandom());
+                builder.sslSocketFactory(sslContext.getSocketFactory(), (X509TrustManager) TRUST_ALL_CERTS);
+            } catch (Exception e) {
+                throw new RuntimeException("Failed to configure SSL for ALLOW_UNTRUSTED", e);
+            }
+        }
```
Comment (jthurne) [inline components/fetch-build-scan-data-cmdline-tool/src/main/java/com/gradle/enterprise/api/client/GradleEnterpriseApiClient.java:84]: 0: Minor, but I would probably extract configuring SSL and configuring proxy authentication into two separate methods.

**Review by jthurne — APPROVED** [review]

**Comment by marcphilipp** [comment]
As discusses, this will be configurable through a script-specific config file instead.

**Comment by etiennestuder** [comment]
@marcphilipp I think @jthurne wanted to rework this PR to use the config file, but maybe he will just start a new one. Not sure. Jim will know :-)

**Comment by marcphilipp** [comment]
@jthurne Feel free to reopen this one if you want. Whatever works best for you.

**Pull request closed without merging**